### PR TITLE
Update typescript to 1.49.3

### PR DESCRIPTION
**Pull request by tsmaeder**
Signed-off-by: Thomas Mäder <tmader@redhat.com>

### What does this PR do?
Updates the typescript plugin to version 1.49.3 

Note. we can't merge this PR until a new image with a new plugin-host executable with '--enableNodeCli' has been built.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16186

### How to test this PR?
Use any of the default typescript default devfiles (for example the angular example) and point it to the meta.yml introduced in this PR.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+displayName: Typescript
+title: Typescript language features
+description: Typescript language features
+icon: https://raw.githubusercontent.com/microsoft/vscode/1.49.3/extensions/typescript-language-features/icon.png
+repository: https://github.com/Microsoft/vscode
+category: Language
+firstPublicationDate: '2020-10-22'
+spec:
+  containers:
+  - image: "quay.io/eclipse/che-sidecar-node:12-026416c"
```
Comment (ericwill) [inline v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml:15]: Let's use the new [integrated sidecar images](https://quay.io/repository/eclipse/che-plugin-sidecar?tag=latest&tab=tags), so `quay.io/eclipse/che-plugin-sidecar:node-c939ba4`

**Review by ericwill — COMMENTED** [review]
Apart from my comment about the sidecar this looks good. Additionally, let's:
1. Merge this PR
2. Update the devfile registry to use this plugin instead of the `che-incubator/typescript` one
3. Open a new PR after the above is merged, and remove `che-incubator/typescript`

**Review by benoitf — DISMISSED** [review]
good work Thomas on the analyze on enableNodeCli issue in nexe and related stuff.

I still have an exception when typescript server is being launched, it displays:
```
ERROR:Remote plugin in vscode-typescriptixe: promise rejection is not handled in two seconds: undefined
ERROR:Remote plugin in vscode-typescriptixe: promise rejection is not handled in two seconds: TypeError: Cannot read property 'stack' of undefined
ERROR:Remote plugin in vscode-typescriptixe: promise rejection stack trace: TypeError: Cannot read property 'stack' of undefined
    at /remote-endpoint/node_modules/@eclipse-che/theia-remote/lib/node/plugin-remote.js:36:25
```
Screenshot 2020-11-19 at 14 17 50

...but I'm able to use code completion/formatting
Screenshot 2020-11-19 at 14 18 30

**Comment by tsmaeder** [comment]
[ci-test-happy-path]

**Review by ericwill — COMMENTED** [review]
Now that we are using the proper extension, can you add an entry for it in the [che-theia-plugins file](https://github.com/eclipse/che-plugin-registry/blob/master/che-theia-plugins.yaml). I think the entry should look like this:

```yaml
- repository:
      url: https://github.com/Microsoft/vscode
      revision: 1.49.3
      directory: extensions/typescript-language-features
    sidecar:
      directory: node
    extensions:
      - https://open-vsx.org/api/vscode/typescript-language-features/1.49.3/file/vscode.typescript-language-features-1.49.3.vsix
```

Otherwise it LGTM.

File: che-theia-plugins.yaml [hunk 2]
```diff
@@ -17,6 +17,14 @@ plugins:
     extensions:
       - https://open-vsx.org/api/ms-vscode/node-debug2/1.42.5/file/ms-vscode.node-debug2-1.42.5.vsix
       - https://open-vsx.org/api/ms-vscode/node-debug/1.44.8/file/ms-vscode.node-debug-1.44.8.vsix
+  - repository:
```
Comment (benoitf) [inline che-theia-plugins.yaml:20]: It's missing an alias for the previous location in che-incubator

File: v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v2
```
Comment (ericwill) [inline v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml:1]: Now that #721 is closed, we do not need this file, correct @benoitf ?

Comment (benoitf) [reply v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml:1]: correct, once aliases and entry are added in che-theia-plugins.yaml ( what this PR dors) we can remove it

Comment (ericwill) [reply v3/plugins/ms-vscode/typescript/1.49.3/meta.yaml:1]: @tsmaeder please remove the meta.yaml changes and we should be good to go.

File: v3/plugins/che-incubator/typescript/1.35.1/meta.yaml [hunk 4]
```diff
@@ -10,6 +10,9 @@ icon: https://raw.githubusercontent.com/microsoft/vscode/1.35.1/extensions/types
 repository: https://github.com/Microsoft/vscode
 category: Programming Languages
 firstPublicationDate: '2019-06-20'
+deprecate:
```
Comment (benoitf) [inline v3/plugins/che-incubator/typescript/1.35.1/meta.yaml:13]: this file should be removed as it will be generated from the che-theia-plugins.yaml file

**Review by benoitf — CHANGES_REQUESTED** [review]

**Review by ericwill — APPROVED** [review]

**Review by benoitf — APPROVED** [review]
Thanks @tsmaeder !

**Pull request merged**